### PR TITLE
Add dragging of layers with the select tool

### LINKED ISSUE #143: Select Tool: Grab/drag selected layers around viewport
If the user has no layers selected, this will select the layer under the cursor (see #140) and then drag (transform) it. If multiple are selected, and the drag occurs on one of the selected, it will drag (transform) all of them. If one or more are selected but the drag occurs on another layer, it deselects the current selection and selects the new layer and drags it (like in the first sentence).

After this is implemented, we can then add support for snapping: #200.

Complexity: 3
Involves: Editor (Rust), Document (Rust)

### PR DESCRIPTION
**Pull request by henryksloan**
Closes #143

- Added a `DragLayer` message to `DocumentMessage` that accepts a path and an offset
- Modified the select tool's FSM to allow:
    - Dragging all selected layers
    - Dragging a single previously unselected layer
    - Drawing a quad and selecting everything within it (by starting the drag on the background)
    - Deselecting everything by clicking the background

---
This change is [Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/293)

### CODE REVIEW

**Review by TrueDoctor — COMMENTED** [review]
Could you add comments where you used the root transform? e.g.
// TODO: Replace with functions of the transform api
this will at least make it easier to fix it afterwards

**Comment by Keavon** [comment]
This looks good to me at a cursory read-through, but the backend code is not my domain of expertise so I defer to True Doctor. Looks like this needs to be synced with master and then you have my blessing to merge if True Doctor is also ok with it.

**Review by Keavon — APPROVED** [review]

**Pull request merged**
